packages/react-core/src/hooks/useIsSignedIn.ts

```diff
@@ -0,0 +1,44 @@
+import { getCurrentUser } from 'aws-amplify/auth';
+import { Hub, HubCallback } from '@aws-amplify/core';
+import { useEffect } from 'react';
+import useDataState, { DataState } from './useDataState';
+
+const action = async (
+  _: { isSignedIn: boolean },
+  input: { setState?: boolean }
+) => {
+  try {
+    await getCurrentUser();
+    return { isSignedIn: true };
+  } catch (error) {
+    if (input?.setState) {
+      return { isSignedIn: false };
+    }
+    throw error;
+  }
+};
+
+export default function useIsSignedIn(): DataState<{ isSignedIn: boolean }> {
+  const [state, handler] = useDataState(action, { isSignedIn: false });
+
+  useEffect(() => {
+    handler();
+
+    const listener: HubCallback = ({ payload }) => {
+      if (payload.event === 'signedIn') {
+        handler();
+      } else if (payload.event === 'signedOut') {
+        handler({ setState: true });
+        handler();
```

Comment:
What is requiring two calls to `handler` here?

Reply:
The first one sets the data and the other sets the error. With useDataState you cannot change both in the same call since the data is set through the return value and message/hasError is set through a thrown error. Is there a better way to do this that you can think of?